### DISCUSSION TITLE
Please update installation guide about wifi

### DISCUSSION THREAD

**Discussion by mtdcy** [osy86/HaC-Mini#569] [Ideas]
There is a M2/NGFF wifi chip named BCM94352Z than can be used to replace original one without adaptor. It works perfect.

**Comment by Medizinmann-MD** [comment]
Also a very good solution are cards using BCMBCM94360NG - also with speeds up to 1200Mbps - and of course also no extra adapter needed.
Availability is good and prices start around $42 on eBay and AliExpress + $10-$15 more on Amazon...

It has been discussed in this thread…
https://github.com/osy/HaC-Mini/issues/191

**Comment by Iaroslav-M** [comment]
So, both BCM94352Z and BCMBCM94360NG don't need any wires? 
--



--

**Comment by Medizinmann-MD** [comment]
> ## So, both BCM94352Z and BCMBCM94360NG don't need any wires?
> --

You don't need an extra adapter NGFF to M.2 as for the cards with BCM94360CS2...
https://osy.gitbook.io/hac-mini-guide/post-installation/wifi#requirements

...so you can still use the 2nd M.2 and no mod on the case is needed.

But of course you must connect it to the WiFi/Bluetooth antennas of you HaC - but you can just use the existing cables - no extra wires needed.

**Comment by immie-jay** [comment]
Please add that Airdrop works as well

**Comment by Medizinmann-MD** [comment]
> Please add that Airdrop works as well

Yepp, AFAIK everything (WiFi, Bluetooth, Airdrop, Handoff etc.) works as these Broadcom-Chipsets are supported natively by MacOS. I personally only use WiFi & Bluetooth (with Magic Trackpad, Headsets etc.) – as I don’t own an iPhone -  so I didn't test Handoff/Airdrop so far.

**Comment by immie-jay** [comment]
Unlocking with Apple Watch or iPhone also works - most times. Appears erratic

I have the BCMBCM94360NG card

**Comment by Medizinmann-MD** [comment]
> Unlocking with Apple Watch or iPhone also works - most times. Appears erratic
> 
> I have the BCMBCM94360NG card

Which MacOS version?

AFAIK there were/are reports about problems with Big Sur in general...?

Correct me if I am wrong!

**Comment by osy** [comment]
Done: https://osy.gitbook.io/hac-mini-guide/post-installation/wifi

**Comment by Ikbosh** [comment]
Does this adapter have better bluetooth support for say AirPod pro's? Referring to the 16Khz/16-Bit

**Comment by keatliang2005** [comment]
anyone experience poor bluetooth range with BCM94352Z ??

with original intel bluetooth range can be quite far, i swap to this it have limited range.
i use some bluetooth earphone, as i get far it will be stuttered... 

i follow the color as labeled. i just wondered could the card wrong ? 
MAIN and AUX

![IMG_0153](https://user-images.githubusercontent.com/473990/114147974-dc25e500-994b-11eb-948c-8f56afdb15ff.jpg)

**Reply by keatliang2005** [reply]
> anyone experience poor bluetooth range with BCM94352Z ??
> 
> with original intel bluetooth range can be quite far, i swap to this it have limited range.
> i use some bluetooth earphone, as i get far it will be stuttered...
> 
> i follow the color as labeled. i just wondered could the card wrong ?
> MAIN and AUX
> 
> ![IMG_0153](https://user-images.githubusercontent.com/473990/114147974-dc25e500-994b-11eb-948c-8f56afdb15ff.jpg)

Follow up, 

hi did anyone with BCM94360NG or BCM94352Z card ?

I get mine from taobao which is Fenvi BCM94352Z, it works out of the box but it seems like there is interference on the bluetooth, result in very short range.

sometime bluetooth earphone will stutter or disconnect.

Wifi work well.
- I tried turn off router 2.4G band, no further improvement
- I tried reverse the IPEX4 connector, only make it worst.
- I tied upload firmware from BrcmPatchRAM kext, it seems like Fenvi BCM94352Z does not support that

1. So i would like to know BCM94360NG have 2.4G band issue ?
2. does anyone with BCM94352Z have the same bluetooth range/connectivity issue ?
3. Could it be the flaw of BCM94352Z, known to have bad bluetooth with interference ?
4. Or i just really bad luck, i got a bad card (broken)
5. Or antenna is broken ?, but previous i using intel wifi with IntelBluetoothFirmware ram patch the range seems fine

**Reply by ninhxuanluu** [reply]
![IMG_4262](https://user-images.githubusercontent.com/8602978/114809161-e0cb1d00-9dd3-11eb-9ace-915e0ecbc886.JPG)
I'm using BCM94360NG and DW1560 having same issue on NUC.
So i try on my PC with Asrock Phantom gaming itx/tb3 and they work fine with stock atena of Asrock motherboard then i think the proplem is the atena of NUC.
btw that issue doesn't happen when i switch to Windows.

**Reply by keatliang2005** [reply]
then probably had to try one of those PCB or flexible antenna for the 2.4Ghz band

**Comment by Medizinmann-MD** [comment]
My range (using bluetooth headsets - Jabra Elite 75t and 65t) with the a card using BCMBCM94360NG is round about 5-10m - never tested the original Intel card though...

**Reply by keatliang2005** [reply]
this is also A+E card fit in nicely right
